### ISSUE TITLE
NavigationView Bug in WinUI 2.6

### ISSUE THREAD

**Issue by zhuxb711** [microsoft/microsoft-ui-xaml#5274]
**Describe the bug**
I'm not quite sure what caused the problem
But this only in "Version2" and Set NavigationView.PaneDisplayMode="Top"
```
ControlsResourcesVersion="Version2"
```
Normal in Version1
![image](https://user-images.githubusercontent.com/42038175/123389918-f4151880-d5cc-11eb-8433-3a4f203a1cfd.png)

BUG in Version2, all the NavigationItem except "Bluetooth Audio" disappeared
![image](https://user-images.githubusercontent.com/42038175/123390468-887f7b00-d5cd-11eb-8a35-4f3a54d53379.png)

I could still found them in Visual Tree
![image](https://user-images.githubusercontent.com/42038175/123390757-d8f6d880-d5cd-11eb-97ea-ef623eb1b99f.png)

If I changed the PaneDisplayMode to "Left" (Hot reload XAML), All items reappear
![image](https://user-images.githubusercontent.com/42038175/123391018-21ae9180-d5ce-11eb-97e0-177bb968c536.png)

If I changed the PaneDisplayMode to "Top" again (Hot reload XAML), but indicator is on the left ???
![image](https://user-images.githubusercontent.com/42038175/123391115-430f7d80-d5ce-11eb-844d-c4c2b8bb03fb.png)

And SettingButton is much wider like this
![image](https://user-images.githubusercontent.com/42038175/123391311-73efb280-d5ce-11eb-963b-7b7e20007927.png)

And the animation is not correct.......
![GIF 2021-6-25 16-01-12](https://user-images.githubusercontent.com/42038175/123391472-a7cad800-d5ce-11eb-8b1b-4a323c27afc2.gif)

I created a blank application and used the same controls. But they work perfectly. So It's very confused, is anything wrong?

If anyone could help me look at this issue, I would be very grateful. Thanks.

**Steps to reproduce the bug**

1. Download the source code from my open source repo [Source code](https://github.com/zhuxb711/RX-Explorer)
2. Go to App.xaml, change ControlsResourcesVersion from "Version1" to "Version2"
![image](https://user-images.githubusercontent.com/42038175/123391885-0bed9c00-d5cf-11eb-8a44-daab56dc0af3.png)
3. Build and run "Package" project (Debug X64) in the solution
4. NavigationView is in [MainPage.xaml Line 353](https://github.com/zhuxb711/RX-Explorer/blob/3068ab13c934854a3b2eef95b795b03d8ab13515/RX_Explorer/View/MainPage.xaml#L353)

**Expected behavior**
Should be same as WinUI 2.5 Or WinUI 2.6 Version1

**Version Info**

NuGet package version:  [Microsoft.UI.Xaml 2.6.0]

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
| Yes |  |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| October 2020 Update (19042)        | Yes   |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Comment by StephenLPeters** [comment]
@zhuxb711 you were unable to recreate the issue in a stand alone application right?

**Comment by StephenLPeters** [comment]
When you click on an item in the LiveVisualTree which isn't displaying on screen is it highlighted? If you inspect its properties do they look normal?

**Comment by zhuxb711** [comment]
@StephenLPeters I find the key code that lead to this issue finally. Please kindly check, thanks.
```csharp
Application.Current.FocusVisualKind = FocusVisualKind.Reveal;
```

Demo: 
[BugTest.zip](https://github.com/microsoft/microsoft-ui-xaml/files/7004159/BugTest.zip)

**Comment by StephenLPeters** [comment]
@gegao18 or @codendone any idea how reveal focus could be causing this?

**Comment by HppZ** [comment]
any update?

**Comment by StephenLPeters** [comment]
fixed by #6317
